api/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaHttpClient.java

```diff
@@ -61,21 +78,41 @@ final class JavaHttpClient implements org.projectnessie.client.http.HttpClient {
     }
 
     client = clientBuilder.build();
+    writerPool = new ForkJoinPool(Math.max(8, ForkJoinPool.getCommonPoolParallelism()));
   }
 
   @Override
   public HttpRequest newRequest(URI baseUri) {
-    return new JavaRequest(this.config, baseUri, client::send);
+    return new JavaRequest(this.config, baseUri, client::send, writerPool);
   }
 
   @Override
   public URI getBaseUri() {
     return config.getBaseUri();
   }
 
+  @SuppressWarnings({"DataFlowIssue", "ResultOfMethodCallIgnored"})
   @Override
   public void close() {
-    client = null;
-    config.close();
+    try {
+      if (client instanceof AutoCloseable) {
+        ((AutoCloseable) client).close();
+      }
+    } catch (Exception e) {
+      // ignore
```

Comment:
Why ignore?

Reply:
not much we could do anyway :shrug:
(the client's also only closeable in more recent Java versions)

Reply:
Why not re-throw? If `config.close()` (below) throws, it propagates :thinking: